### PR TITLE
QOL changes for PercentHealth module

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: src/main/java/gg/uhc/uhc/modules/health/PercentHealthObjectiveModule.java [hunk 1]
```diff
@@ -211,6 +222,27 @@ public void on(PlayerQuitEvent event) {
...
 
+    @EventHandler(priority = EventPriority.MONITOR, ignoreCancelled = true)
+    public void on(EntityDamageEvent event) {
+        Entity entity = event.getEntity();
+        if (!(entity instanceof Player)) {
+            return;
+        }
+
+        Player player = (Player) entity;
+        // Calculate new health based on final damage instead of waiting for one tick
```
Comment (Eluinhost) [inline src/main/java/gg/uhc/uhc/modules/health/PercentHealthObjectiveModule.java:241]: If we wait one tick here and remove ignoreCancelled=true we can take into account damage events that cancel and set the health manually

File: docs/modules/PercentHealth.md [hunk 2]
```diff
@@ -20,7 +21,7 @@ objectives:
   scaling: 5
 ```
 
-`update period` - how often to update player's health (in ticks)  
+`update period` - how often to forcibly update all online player's health (in ticks)
```
Comment (Eluinhost) [inline docs/modules/PercentHealth.md:24]: If we're going to increase the force refresh time to 5 seconds from 1 we should probably add a note to say that for plugins that set health manually without damaging the player a lot the number should be lowered to keep it responsive.

**Comment by Eluinhost** [comment]
Also could you squash your commits before I merge.

**Comment by ghost** [comment]
Squashed everything and did everything we talked about privately

**Comment by Eluinhost** [comment]
@D4mnX  I just noticed this PR didn't have the 5 seconds change in it, was it lost in the squash or did I just imagine it?

**Comment by ghost** [comment]
I believe I force-push changed it from 5 seconds back to 1 as we agreed privately that performance would not be an issue.

**Comment by Eluinhost** [comment]
OK that makes sense, thanks

**Pull request merged**